### ISSUE TITLE
Occuring an unknown error while padding state to the command

### ISSUE THREAD

**Issue by rdwnsjjd** [tauri-apps/tauri#3087]
I got this error while passing my structure as state to my command:
```
thread 'main' panicked at 'container::get(): get() called before set() for given type', .cargo/registry/src/github.com-1ecc6299db9ec823/state-0.5.2/src/container.rs:603:14
```

my command:
```rust
#[tauri::command]
pub fn apply(
    data: String,
    my_struct: tauri::State<'_, Arc<Mutex<Option<MyStruct>>>>
) -> Result<(), String> {
```

my info: 
```
Operating System - Ubuntu, version 20.04 X64

Node.js environment
  Node.js - 14.18.1
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - 1.0.0-beta.8

Global packages
  npm - 8.1.3
  yarn - 1.22.17

Rust environment
  rustc - 1.52.1
  cargo - 1.52.0

App directory structure
/src-tauri
/src
/node_modules
/public

App
  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../src
  devPath - http://localhost:3000
  framework - React

```

What is it?

**Comment by FabianLars** [comment]
This means you're trying to access the state before you gave it to tauri. Aka you called app.state() (this is the same as having the state as a command argument) before app.manage().
This is not a bug but expected behavior (unless you _did_ call manage before)

**Comment by rdwnsjjd** [comment]
So why does it cause it at all?
This is my code:
```rust
fn main() {
    tauri::Builder::default()
        .manage(|| {
            // somthing stuff
        })
        .invoke_handler(tauri::generate_handler![apply])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}

```

I call `manage` before `invoke_handler`

**Comment by FabianLars** [comment]
tl;dr: give manage the actual object not just a closure which won't get executed.

Okay i don't know if github is trolling me or you just deleted your comment because it doesn't show me anything here, but our discord bot shows it, so i'll answer it anyway lol.
So just in case you actually deleted it here is what you wrote:
```rs
fn main() {
    tauri::Builder::default()
        .manage(|| {
            // somthing stuff
        })
        .invoke_handler(tauri::generate_handler![apply])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

The way State _finds_ your managed stuff is basically by comparing Signatures, that's why you can only store one instance per type. The problem here is you're asking for `Arc<Mutex<Option<MyStruct>>>`, but you gave the Manager a closure (i guess of type: `|| -> Arc<Mutex<Option<MyStruct>>>`). This is not the same, because the closure won't be executed/resolved _automatically_.
The way `manage` is intended to be used is to give it the actual object **directly** or by **calling** a function (which returns the object), not just giving it a function which won't ever be executed.
